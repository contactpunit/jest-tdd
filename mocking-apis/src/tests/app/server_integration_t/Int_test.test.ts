import { HTTP_CODES, HTTP_METHODS } from "../../../app/model/ServerModel";
import { Server } from "../../../app/server/Server";
import { makeAwesomeRequest } from "./http_client";

describe('Integration test for Server GET POST calls', () => {

    let server: Server
    beforeAll(async() => {
        server = new Server()
        await server.startServer()
    })

    afterAll(async() => {
        await server.stopServer()
    })

    const someUser = {
        id: '',
        userName: 'someuser',
        password: 'somepass'
    }

    it('should register the user using custom http client lib', async () => {
        const result = await makeAwesomeRequest({
            host: 'localhost',
            port: 8080,
            method: HTTP_METHODS.POST,
            path: '/register'

        }, someUser)

        expect(result.statusCode).toBe(HTTP_CODES.CREATED)
        expect(result.body.userId).toBeDefined()
        
    })
})