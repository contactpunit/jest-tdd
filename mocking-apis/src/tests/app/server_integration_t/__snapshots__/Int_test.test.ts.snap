// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test for Server GET POST calls should ceate snapshot of the reservation after creation 1`] = `
{
  "endDate": "Mon Dec 09 2024",
  "id": "123456",
  "room": "123",
  "startDate": "Mon Dec 09 2024",
  "user": "punit",
}
`;

exports[`Integration test for Server GET POST calls should ceate snapshot of the reservation after creation 2`] = `
{
  "endDate": "Mon Dec 09 2024",
  "id": "123456",
  "room": "123",
  "startDate": "Mon Dec 09 2024",
  "user": "punit",
}
`;
